server/lib/request-handler.js

```diff
@@ -1,9 +1,70 @@
-import { wordAssocAPIk } from './api_config.js';
+//import { wordAssocAPIk } from './api_config.js';
+var apiKey = require('./api_config.js');
 var path = require('path');
 var unirest = require('unirest');
-var db = require('/../../db');
+var User = require('./../../db/models/user');
+var util = require('./utility.js');
+//var db = require('/../../db');
 
+/////////////////////////////////////////////////////////////////
+//AUTHENTICATION
 
+exports.userLogin = function(req, res) {
+  var username = req.body.username;
+  var password = req.body.password;
+
+  User.findOne( {username: username} )
+    .exec(function(err, user) {
+      if(!user) {
+        res.writeHead(401);
+        res.end();
+      } else {
+        util.comparePassword(password, user.password, function(err, match) {
+          if (match) {
+            util.createSession(req, res, user);
+          } else {
+            res.writeHead(401);
+            res.end();
+          }
+        });
+      }
+    });
+};
+
+exports.userLogout = function(req, res) {
+  req.session.destroy(function() {
+    res.writeHead(200);
+    res.end();
+  });
+};
+
+
+exports.userSignup = function(req, res) {
+  var username = req.params.username;
+  var password = req.body['password'];
+  var userEmail = req.body['email'];
+  var userAddress = req.body['address'];
+  var streetAddress = userAddress['houseNum'];
+  streetAddress = streetAddress.toString();
+  streetAddress = streetAddress.concat(
+    userAddress['streetName'],
+    ', ',
+    userAddress['city'],
+    ', ',
+    userAddress['state']
+    );
+
+  util.geoCodeit(res, streetAddress, function(err, response) {
+    if (!err) {
+      var geoLocation = response.json.results[0].geometry.location;
+      res.send(geoLocation);
```

Comment:
is the geoLocation object identical to the one we discussed eg {location: {lat: <lat>, long: <long>}}, ? Google Map has a convention writing the longitude as the 'lng' property.